pipelines/infrastructure/ol_infrastructure/instance_refresh_concourse_ami_ci.yaml

```diff
@@ -0,0 +1,96 @@
+---
+resource_types:
+- name: ami
+  type: docker-image
+  source:
+    repository: jdub/ami-resource
+- name: pulumi
+  source:
+    repository: mitodl/concourse-pulumi-resource-provisioner
+  type: docker-image
+
+resources:
+- name: concourse-worker-ami
+  type: ami
+  check_every: 1m
+  source:
+    region: us-east-1
+    filters:
+      owner-id: 610119931565
+      is-public: false
+      state: available
+      name: concourse-worker-*
+
+- name: concourse-web-ami
+  type: ami
+  check_every: 1m
+  source:
+    region: us-east-1
+    filters:
+      owner-id: 610119931565
+      is-public: false
+      state: available
+      name: concourse-web-*
+
+- icon: github
+  name: pulumi_files
+  source:
+    branch: issue_679
+    paths:
+    - src/ol_infrastructure/applications/concourse/
+    uri: https://github.com/mitodl/ol-infrastructure
+  type: git
+
+- name: pulumi-infra
+  source:
+  type: pulumi
+
+jobs:
+- name: concourse-ami-build
+  plan:
+  - task: set-aws-creds
+    config:
+      platform: linux
+      image_resource:
+        type: registry-image
+        source:
+          repository: amazon/aws-cli
+          tag: 'latest'
+      outputs:
+      - name: aws_creds
+      run:
+        path: sh
+        args:
+        - -exc
+        - |
+          AWS_ROLE=$(aws sts get-caller-identity | grep 'arn' |  perl -ne 'print $1 if  /(arn.*)\//')
+          export $(printf "AWS_ACCESS_KEY_ID=%s AWS_SECRET_ACCESS_KEY=%s AWS_SESSION_TOKEN=%s" \
+          $(aws sts assume-role \
+          --role-arn $AWS_ROLE \
+          --role-session-name MySessionName \
+          --query "Credentials.[AccessKeyId,SecretAccessKey,SessionToken]" \
+          --output text))
+          cd aws_creds
+          echo "[default]" > credentials
+          echo "aws_access_key_id=$AWS_ACCESS_KEY_ID" >> credentials
+          echo "aws_secret_access_key=$AWS_SECRET_ACCESS_KEY" >> credentials
+          echo "aws_session_token=$AWS_SESSION_TOKEN" >> credentials
```

Comment:
It's probably worth pulling this script into its own file in the repository so we can use it across all of our pipelines without having to copy/paste it each time. Then we can update these three pipelines to use the same script.

Reply:
hmm is there a way how to include a script in concourse? include statement?

Reply:
You would just pass the path of the script in the repository as the parameter to the `run` argument of the task step. So, rather than in-lining the script as you did above with the `run` parameter being `sh` it would just be the path to the script as it exists in the repository.